### PR TITLE
#407 Allow Removal Of A Payment Method.

### LINKED ISSUE #407: Allow Removal Of A Payment Method
The user should be able to delete a PaymentMethod from their Real Wallet. The 'Delete' icon should only be visible when the method is inactive (we cannot remove an active PaymentMethod). 

The backend endpoint for this also needs to be implemented. We can use simply ``PaymentMethod.remove()`` ( ``StripePaymentMethod.remove()`` will also remove it from Stripe).

### PR DESCRIPTION
**Pull request by criske**
closes #407

Self XDSD

### CODE REVIEW

**Comment by zoeself** [comment]
@criske thank you for your Pull Request. I'll assign someone to review it soon.

File: src/main/java/com/selfxdsd/selfweb/api/PaymentMethodsApi.java [hunk 1]
```diff
@@ -289,6 +294,60 @@ public PaymentMethodsApi(final User user) {
...
+        );
+        ResponseEntity<String> response;
+        try {
+            final PaymentMethod paymentMethod = this
+                .getStripePaymentMethod(owner, name, paymentMethodId);
+            if (!paymentMethod.active()) {
+                final PaymentMethods paymentMethods = paymentMethod
+                    .wallet()
+                    .paymentMethods();
+                final boolean removed = paymentMethods.remove(paymentMethod);
```
Comment (amihaiemil) [inline src/main/java/com/selfxdsd/selfweb/api/PaymentMethodsApi.java:328]: @criske You have to use ``PaymentMethod.remove()`` here. Look inside StripePaymentMethod.remove(), that method will remove it from Stripe first, before removing it from our DB.

**Review by amihaiemil — CHANGES_REQUESTED** [review]
@criske looks good, just one comment : D

File: src/main/java/com/selfxdsd/selfweb/api/PaymentMethodsApi.java (hunk 1 above)
Comment (criske) [reply src/main/java/com/selfxdsd/selfweb/api/PaymentMethodsApi.java]: @amihaiemil  sorry about that :D I also noticed that I removed a test by accident.

**Comment by amihaiemil** [comment]
@criske looks good, thanks!

**Comment by amihaiemil** [comment]
@rultor merge it

**Pull request merged**